lib/player.js

```diff
@@ -5725,16 +5725,20 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
     if (this.loadMode_ == shaka.Player.LoadMode.MEDIA_SOURCE) {
       // Filter by current videoId and has audio.
       filteredTracks = variants.filter((t) => {
-        return t.originalVideoId === active.originalVideoId && t.audioCodec;
+        return t.originalVideoId === active.originalVideoId &&
+            t.videoId === active.videoId && t.audioCodec;
       });
     }
     if (!filteredTracks.length) {
       return [];
     }
 
-    /** @type {!Set<shaka.extern.AudioTrack>} */
-    const audioTracksSet = new Set();
+    /** @type {!Map<number, shaka.extern.AudioTrack>} */
+    const audioTracksMap = new Map();
```

Comment:
Please make a patch with this for current releases!

Reply:
I created https://github.com/shaka-project/shaka-player/pull/8460

Reply:
I reverted the changes on this PR